app/oauth.go

```diff
@@ -896,8 +896,12 @@ func (a *App) AuthorizeOAuthUser(w http.ResponseWriter, r *http.Request, service
 		mlog.Error("Error getting OAuth user", mlog.Int("response", resp.StatusCode), mlog.String("body_string", bodyString))
 
 		if service == model.ServiceGitlab && resp.StatusCode == http.StatusForbidden && strings.Contains(bodyString, "Terms of Service") {
+			url, err := url.Parse(*sso.UserAPIEndpoint)
+			if err != nil {
+				return nil, "", stateProps, nil, model.NewAppError("AuthorizeOAuthUser", "api.user.authorize_oauth_user.service.app_error", map[string]any{"Service": service}, err.Error(), http.StatusInternalServerError)
+			}
```

Comment:
Is there a need to parse this again? For simplicity we could just pass the full URL.

Reply:
@agnivade We are parsing the url because usually sso.UserAPIEndpoint have value as "https://example.com/api/v4/user" .Here the hardcode value was github.com to make this consistent we sending the url.Hostname here. If we directly send the error then it would look something like this 
```
GitLab's Terms of Service have updated. Please go to https://example.com/api/v4/user to accept them and then try logging into Mattermost again.
```
which is not consistent with original version

Reply:
That makes sense. Thanks.